### PR TITLE
Improve helper functions in vm_developer_common_widgets

### PR DESCRIPTION
**Pull request by migueltylee**
Refactors and adds functionality to helper functions in  vm_developer_common_widgets.dart, mainly the _objectName and qualifiedName (previously _ownerName) functions.
These improvements will be helpful for the display of the remaining object types in the object inspector.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 1]
```diff
@@ -177,41 +177,50 @@ class RequestableSizeWidget extends StatelessWidget {
...
+  } else {
+    objectRefName = objectRef?.vmType;
   }
 
   return objectRefName;
 }
 
-/// Recursively gets the full owner name of a field or function object.
-String? _ownerName(ObjRef? ref) {
+/// Recursively gets the name of a function, qualified with its full owner name,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:198]: what does 'qualified' mean in this context? can you elaborate?

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 2]
```diff
@@ -334,9 +343,11 @@ class RetainingPathWidget extends StatelessWidget {
         for (RetainingObject object in retainingPath.elements!.sublist(1))
           Row(
             children: [
-              SelectableText(
-                _retainingObjectDescription(object),
-                style: Theme.of(context).fixedFontStyle,
+              Flexible(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:346]: why do all these rows only have a single child? makes me wonder why we need the rows at all

**Review by kenzieschmoll — APPROVED** [review]
generally lgtm with a couple comments

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 1 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:198]: For example: for the function build with owner GalleryApp, the qualified name would be GalleryApp.build. 
These are some other examples:
image

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 2 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:346]: I think it is so the contents can be flexible, when the text is too long

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:198]: can you add this context to the dart doc? thanks

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 3]
```diff
@@ -177,41 +177,50 @@ class RequestableSizeWidget extends StatelessWidget {
...
 
-  if (objectRef is ClassRef ||
-      objectRef is FuncRef ||
-      objectRef is FieldRef ||
-      objectRef is LibraryRef) {
+  if (objectRef is ClassRef || objectRef is FuncRef || objectRef is FieldRef) {
     objectRefName = (objectRef as dynamic).name;
+  } else if (objectRef is LibraryRef) {
+    objectRefName =
+        objectRef.name?.isEmpty ?? false ? objectRef.uri : objectRef.name;
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:185]: I'm not sure if we have a lint against this, but I generally wrap non-trivial conditions with parens to improve readability. Something like this:

```dart
objectRefName = (objectRef.name?.isEmpty ?? false) ? objectRef.uri : objectRef.name;
```

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 4]
```diff
@@ -177,41 +177,50 @@ class RequestableSizeWidget extends StatelessWidget {
...
+    return '${ref.name}';
+  } else if (ref is FuncRef) {
+    if (ref.owner is! LibraryRef) {
+      return '${qualifiedName(ref.owner)}.${ref.name}';
+    } else {
+      return '${ref.name}';
+    }
   }
+
+  return '';
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:213]: Should we ever get to this case? If we don't expect to, maybe consider throwing an exception.

**Review by bkonyi — APPROVED** [review]
LGTM with minor comments.

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 3 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:185]: It doesn't show in the dart analysis, but I'll add the parenthesis.

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 1 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:198]: Will do

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 5]
```diff
@@ -177,41 +177,56 @@ class RequestableSizeWidget extends StatelessWidget {
...
   }
 
   return objectRefName;
 }
 
-/// Recursively gets the full owner name of a field or function object.
-String? _ownerName(ObjRef? ref) {
+/// Returns the name of a function, qualified with the name of
+/// its owner added as a prefix, separated by a period.
+/// For example: for function build with owner class Foo,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:200]: dart doc nit: always separate the first sentence of dart doc from the rest with a blank line
```
/// my first sentence blah blah blah.
///
/// the rest. of the sentences. blah blah.
```

**Pull request merged**